pkg/plugins/runtime/k8s/webhooks/injector/injector.go

```diff
@@ -213,13 +232,83 @@ func (i *KumaInjector) meshFor(pod *kube_core.Pod, ns *kube_core.Namespace) (*co
 	return meshResource, nil
 }
 
-func (i *KumaInjector) namespaceFor(pod *kube_core.Pod) (*kube_core.Namespace, error) {
+type namedContainerPatches struct {
+	names   []string
+	patches []mesh_k8s.JsonPatchBlock
+}
+
+// loadContainerPatches loads the ContainerPatch CRDs associated with the given pod, divides out
+// the sidecar and init patches, and concatenates each type into its own list for return.
+func (i *KumaInjector) loadContainerPatches(
+	ctx context.Context,
+	logger logr.Logger,
+	pod *kube_core.Pod,
+	ns *kube_core.Namespace,
+) (sidecarPatches namedContainerPatches, initPatches namedContainerPatches, err error) {
+	patchNames := i.cfg.ContainerPatches
+	if val, exist := metadata.Annotations(pod.Annotations).GetString(metadata.KumaContainerPatches); exist {
+		for _, patchName := range strings.Split(val, ",") {
+			if patchName != "" {
+				patchNames = append(patchNames, patchName)
+			}
+		}
+	}
+
+	for _, patchName := range patchNames {
+		containerPatch := &mesh_k8s.ContainerPatch{}
+		if err := i.client.Get(ctx, kube_types.NamespacedName{Namespace: ns.GetName(), Name: patchName}, containerPatch); err != nil {
+			// skip the patch rather than fail.
+			logger.Error(err, "invalid ContainerPatch, skipping.", "name", patchName)
```

Comment:
Hmm the design doc was mentioning a failure in this case no?

Reply:
Can't find this discussion in the proposal :( can you link to it?
I feel it might be a bit dangerous, remove ContainerPatch and nothing is injecting anymore. On the other hand, you can easily recreate it, even an empty ContainerPatch...

Reply:
https://github.com/kumahq/kuma/blob/master/docs/proposals/custom-sidecar-config.md#error-modes-and-validation

>  If a workload refers to a ContainerPatch which does not exist, the injection will explicitly fail and log the failure.

Reply:
changed